app/controllers/users/registrations_controller.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+require 'mail'
+
+class Users::RegistrationsController < Devise::RegistrationsController
+  before_action :redirect_if_university, only: [:create]
+
+  protected
+
+    def redirect_if_university
+      Rails.logger.info(Mail::Address.new(params[:user][:email]).domain.to_s)
```

Comment:
set a `domain` variable rather than call `Mail::Address.new` twice (for the logging and then the case-statement).

Also, if this logging line is to be kept in the code, can you add some context about or detail of what is happening so the log includes something meaningful to read later.